docs/Doxyfile.in

```diff
@@ -602,12 +602,6 @@ HIDE_SCOPE_NAMES       = NO
 
 HIDE_COMPOUND_REFERENCE= NO
 
-# If the SHOW_HEADERFILE tag is set to YES then the documentation for a class
-# will show which file needs to be included to use the class.
-# The default value is: YES.
-
-SHOW_HEADERFILE        = YES
```

Comment:
What's the reasoning behind removing this?

Reply:
This produces warnings in the build like 
```
warning: ignoring unsupported tag 'SHOW_HEADERFILE' at line 609, file Doxyfile
warning: ignoring unsupported tag 'WARN_IF_INCOMPLETE_DOC' at line 827, file Doxyfile
warning: ignoring unsupported tag 'FULL_SIDEBAR' at line 1574, file Doxyfile
warning: ignoring unsupported tag 'MATHJAX_VERSION' at line 1657, file Doxyfile
```

I think this Doxyfile probably came from a newer version of Doxygen than we use in CI. I guess I don't have a strong opinion about keeping them in or not.